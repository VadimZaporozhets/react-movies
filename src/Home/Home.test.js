import React from 'react';
import { HomeSceneComponent } from './index';
import { shallow } from 'enzyme';
import '@babel/polyfill';

describe('<HomeSceneComponent />', () => {
    let props, wrapper;

    beforeEach(() => {
        props = {
            classes: {
                home: {},
                progress: {}
            }
        };

        wrapper = shallow(<HomeSceneComponent {...props} />);
    });

    it('should render home component with loader', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render home component with movie tiles pane', () => {
        const movies = [
            {
                id: 447365,
                title: 'Guardians of the Galaxy Vol. 3',
                poster_path:
                    'https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg',
                releaseYear: '2020',
                genres: ['Action', 'Adventure', 'Science Fiction']
            },
            {
                id: 424785,
                title: 'Transformers 7',
                poster_path:
                    'https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg',
                releaseYear: '2019',
                genres: ['Science Fiction', 'Action', 'Adventure']
            }
        ];

        wrapper.setState({
            movies
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('componentDidMount should load movies', async () => {
        await wrapper.instance().componentDidMount();

        expect(wrapper.state().movies).toContainEqual(
            expect.objectContaining({
                id: expect.any(Number),
                title: expect.any(String),
                poster_path: expect.any(String),
                releaseYear: expect.any(String),
                genres: expect.any(Array)
            })
        );
    });
});
