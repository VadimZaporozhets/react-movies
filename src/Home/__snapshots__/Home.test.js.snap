// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeSceneComponent /> should render home component with loader 1`] = `
<main
  className={Object {}}
>
  <WithStyles(SearchPanelComponent) />
  <WithStyles(SortResultsPanelComponent) />
  <WithStyles(CircularProgress)
    className={Object {}}
  />
</main>
`;

exports[`<HomeSceneComponent /> should render home component with movie tiles pane 1`] = `
<main
  className={Object {}}
>
  <WithStyles(SearchPanelComponent) />
  <WithStyles(SortResultsPanelComponent) />
  <MovieTilesPane
    movies={
      Array [
        Object {
          "genres": Array [
            "Action",
            "Adventure",
            "Science Fiction",
          ],
          "id": 447365,
          "poster_path": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
          "releaseYear": "2020",
          "title": "Guardians of the Galaxy Vol. 3",
        },
        Object {
          "genres": Array [
            "Science Fiction",
            "Action",
            "Adventure",
          ],
          "id": 424785,
          "poster_path": "https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg",
          "releaseYear": "2019",
          "title": "Transformers 7",
        },
      ]
    }
    title="Found results:"
  />
</main>
`;
